src/lib.rs

```diff
@@ -242,6 +251,31 @@ impl<T: ContactInfo> RoutingTable<T> {
         }
     }
 
+    /// Removes `name` from routing table and returns `false` if we no longer need to stay connected.
+    ///
+    /// We should remain connected iff entry at bucket index of `name` is in the routing table and is
+    /// within the `GROUP_SIZE` closest nodes in that bucket.
+    pub fn need_to_keep(&mut self, name: &XorName) -> bool {
```

Comment:
`need_to_keep` sounds like it doesn't mutate the table; since it (possibly) actually removes the entry, I'd rather call it e. g. `remove_if_unneeded`.